examples/consumer_offset_metadata/consumer_offset_metadata.go

```diff
@@ -0,0 +1,99 @@
+// Example Apache Kafka consumer that commit offset with metadata
+package main
+
+/**
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// consumer_offset_metadata implements a consumer that commit offset with metadata that represents the state
+// of the partition consumer at that point in time. The metadata string can be used by another consumer
+// to restore that state, so it can resume consumption.
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+)
+
+func main() {
+
+	if len(os.Args) < 6 {
+		fmt.Fprintf(os.Stderr, "Usage: %s <broker> <group> <topic> <partition> <offset> \"<metadata>\"\n",
```

Comment:
How about having two modes:
 * if offset and metadata specified: commit and show committed offset
 * if offset and metadata not specified: just show the committed offset

I think that would make it even more useful.

Reply:
Good idea! Done, PTAL.